Java/core/src/main/java/com/amazon/randomcutforest/AbstractForestTraversalExecutor.java

```diff
@@ -192,4 +192,22 @@ public void update(double[] point) {
      */
     public abstract <R, S> S traverseForestMulti(double[] point,
             Function<RandomCutTree, MultiVisitor<R>> visitorFactory, Collector<R, ?, S> collector);
+
+    /**
+     * Returns a clean deep copy of the point.
+     *
+     * Current clean-ups include changing negative zero -0.0 to positive zero 0.0.
+     *
+     * @param point The original data point.
+     * @return a clean deep copy of the original point.
+     */
+    protected double[] cleanCopy(double[] point) {
+        double[] pointCopy = Arrays.copyOf(point, point.length);
+        for (int i = 0; i < point.length; i++) {
+            if (pointCopy[i] == 0.) {
```

Comment:
Elsewhere in this code base we're using explicit `.0`. Can we write it out here for consistency?

Reply:
yes, i have changed them to 0.0